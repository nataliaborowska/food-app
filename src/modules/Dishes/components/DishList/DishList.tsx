import React from 'react';

import {Dish} from './Dish';
import {IDish} from '../../../../store/reducers/dishesReducer';

interface IPropTypes {
  dishes: Array<IDish>;
}

export const DishList: React.FC<IPropTypes> = (props) => (
  <tbody>
    {props.dishes.map(dish => (
      <Dish
        key={dish._id}
        name={dish.name}
        description={dish.description}
        size={dish.size}
        type={dish.type}
        price={dish.price}
      />
    ))}
  </tbody>
)