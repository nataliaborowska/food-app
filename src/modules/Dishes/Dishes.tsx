import React, {useEffect, useState} from 'react';
import axios from 'axios';

import {DishList} from './DishList';

export const Dishes: React.FC = () => {
  const [dishes, setDishes] = useState([]);
  const [fetching, setFetching] = useState(true);

  useEffect(() => {
    axios.get('http://localhost:5000/dishes')
      .then(response => setDishes(response.data))
      .catch(error => {console.log(error)})
      .finally(() => {
        setFetching(false)
      })
  }, []);

  if (fetching) {
    return (
      <span>Loading ...</span>
    )
  }

  return (
    <div>
      <table className="table">
        <thead>
          <tr>
            <th scope="col">Name</th>
            <th scope="col">Description</th>
            <th scope="col">Size</th>
            <th scope="col">Price</th>
          </tr>
        </thead>

        {dishes ?
          <DishList dishes={dishes} />
          :
          <tbody>
            <tr>
              <td>Sorry, no dishes found</td>
            </tr>
          </tbody>
        }
      </table>
    </div>
  )
}