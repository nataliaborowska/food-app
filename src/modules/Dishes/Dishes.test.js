import {mount, shallow} from 'Enzyme';
import axios from 'axios';
import {act} from 'react-dom/test-utils';

import {Dishes} from './Dishes';
import {DishList} from './DishList';

describe('testing Dishes', () => {
  it('check if dishes render correctly', () => {
    const wrapper = shallow(<Dishes />);

    expect(wrapper).toMatchSnapshot();
  });
});

jest.mock('axios');

const dishesArray = [{
  name: 'test',
  description: 'test',
  size: 2,
  price: 2,
  _id: 'test',
}];

const data = {
  data: dishesArray,
};

describe('test loading dishes', () => {
  let wrapper;

  afterEach(() => jest.clearAllMocks());

  it('renders with loading', () => {
    wrapper = shallow(<Dishes />);

    expect(wrapper.find('span').first().text()).toBe('Loading ...');
  });

  it('loads dishes', async () => {
    await act(async () => {
      await axios.get.mockImplementationOnce(() => Promise.resolve(data))
      wrapper = mount(<Dishes />)
    });

    wrapper.update();

    await expect(axios.get).toHaveBeenCalledWith('http://localhost:5000/dishes');

    await expect(axios.get).toHaveBeenCalledTimes(1);

    await expect(wrapper.find(DishList).props().dishes.sort()).toEqual(dishesArray.sort())
  })
})