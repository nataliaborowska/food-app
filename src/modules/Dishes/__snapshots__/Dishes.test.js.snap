// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Dishes check if dishes render correctly 1`] = `
<span>
  Loading ...
</span>
`;
